pkg/compiler/rego/rego.go

```diff
@@ -185,10 +185,14 @@ func (c *CompilerRego) compileCondition(o ast.Condition, lvl, lineNum int) (stri
 		if strings.HasPrefix(id, "ctx.") {
 			id = strings.Replace(id, "ctx", "input", 1)
 		}
+		if strings.HasPrefix(id, types.SUBJECT+".") {
+			id = strings.Replace(id, types.SUBJECT, "seal_subject", 1)
+		}
+
 		return id, nil
 
 	case *ast.PrefixCondition:
-		rhs, err := c.compileCondition(s.Right, lvl+1, lineNum)
+		rhs, err := c.compileCondition(s.Right, lvl, lineNum)
```

Comment:
these are recursive calls - please do not remove `lvl+1` here and elsewhere in the function

if spaces() needs to work, then that needs to accommodate the passing of lvl+1 on recursive calls

Reply:
Got it, reverted. 
Plus added prettify method to make rego much more pretty

Reply:
thank you